ocean_lib/models/v4/test/test_erc20_token.py

```diff
@@ -0,0 +1,403 @@
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    assert registered_event[0].event == "NFTCreated"
+    assert registered_event[0].args.admin == publisher_wallet.address
+    token_address = registered_event[0].args.newTokenAddress
+    erc721_token = ERC721Token(web3, token_address)
+    assert erc721_token.contract.caller.name() == "DT1"
+    assert erc721_token.symbol() == "DTSYMBOL"
+
+    # Tests current NFT count
+    current_nft_count = erc721_factory.get_current_nft_count()
+    erc721_factory.deploy_erc721_contract(
+        "DT2",
+        "DTSYMBOL1",
+        1,
+        ZERO_ADDRESS,
+        "https://oceanprotocol.com/nft/",
+        publisher_wallet,
+    )
+    assert erc721_factory.get_current_nft_count() == current_nft_count + 1
+
+    # Tests get NFT template
+    nft_template_address = get_address_of_type(config, ERC721Token.CONTRACT_NAME, "1")
+    nft_template = erc721_factory.get_nft_template(1)
+    assert nft_template[0] == nft_template_address
+    assert nft_template[1] is True
+
+    # Tests creating successfully an ERC20 token
+    erc721_token.add_to_create_erc20_list(consumer_wallet.address, publisher_wallet)
+    erc_create_data = ErcCreateData(
+        1,
+        ["ERC20DT1", "ERC20DT1Symbol"],
+        [
+            publisher_wallet.address,
+            consumer_wallet.address,
+            publisher_wallet.address,
+            ZERO_ADDRESS,
+        ],
+        [web3.toWei("0.5", "ether"), 0],
+        [b""],
+    )
+    tx_result = erc721_token.create_erc20(erc_create_data, consumer_wallet)
+    assert tx_result, "Failed to create ERC20 token."
+
+    tx_receipt = web3.eth.wait_for_transaction_receipt(tx_result)
+    registered_token_event = erc721_factory.get_event_log(
+        ERC721FactoryContract.EVENT_TOKEN_CREATED,
+        tx_receipt.blockNumber,
+        web3.eth.block_number,
+        None,
+    )
+    assert registered_token_event, "Cannot find TokenCreated event."
+
+    # Tests templateCount function (one of them should be the Enterprise template)
+    assert erc721_factory.template_count() == 2
+
+    # Tests ERC20 token template list
+    erc20_template_address = get_address_of_type(config, ERC20Token.CONTRACT_NAME, "1")
+    template = erc721_factory.get_token_template(1)
+    assert template[0] == erc20_template_address
+    assert template[1] is True
+
+    # Create an ERC20 with publish Fees ( 5 USDC, going to publishMarketAddress)
+    erc721, erc20 = deploy_erc721_erc20(
+        web3=web3,
+        config=config,
+        erc721_publisher=publisher_wallet,
+        erc20_minter=publisher_wallet,
+        cap=web3.toWei(publishMarketFeeAmount, "ether"),
+    )
+
+    # Check erc20 params
+    assert erc20.get_id() == 1
+    assert erc20.contract.caller.name() == "ERC20DT1"
+    assert erc20.symbol() == "ERC20DT1Symbol"
+    assert erc20.decimals() == 18
+    assert erc20.cap() == web3.toWei(publishMarketFeeAmount, "ether")
```

Comment:
can we replace with snake case or put the actual value here if it's used one time?